Fix Qwen2VL processor to handle odd number of frames

jla524 opened the pull request:
# What does this PR do?

Fixes #35412 (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?


## Who can review?

@ArthurZucker, @amyeroberts, @qubvel

jla524 committed:
fix: processing odd number of frames

jla524 committed:
feat: add test case

jla524 committed:
update: test one frame

zucchini-nlp reviewed (commented):
Hey @jla524 !

Thanks for fixing! I think we should not hardcode `2` here as the temporal patch size, as the patch size is configurable when initializing the image processor. Can you generalize the logic for all temporal patch sizes?

jla524 committed:
feat: support custom patch size

jla524 commented:
> Hey @jla524 !
> 
> Thanks for fixing! I think we should not hardcode `2` here as the temporal patch size, as the patch size is configurable when initializing the image processor. Can you generalize the logic for all temporal patch sizes?

Thanks for reviewing! Yes, it should use the actual value of the temporal patch size. I have fixed it and added a test.

zucchini-nlp reviewed (commented):
Thanks for iterating! It doesn't seem to be the right solution for me yet, left some comments

jla524 committed:
fix: test with videos

jla524 committed:
revert: change on patch repeat

jla524 committed:
fix: much wow

jla524 committed:
update: fixups

jla524 committed:
fixup pls

jla524 committed:
ruff fixup

jla524 committed:
Merge branch 'main' into fix/qwen2vl_processor

jla524 commented on tests/models/qwen2_vl/test_image_processing_qwen2_vl.py:
```diff
@@ -247,3 +247,27 @@ def test_nested_input(self):
+            prcocess_out = image_processing(image_inputs, return_tensors="pt")
+            encoded_images = prcocess_out.pixel_values
+            expected_output_image_shape = (expected_dims, 1176)
+            self.assertEqual(tuple(encoded_images.shape), expected_output_image_shape)
```
updated tests to use video inputs

jla524 commented on tests/test_image_processing_common.py:
```diff
@@ -125,19 +125,19 @@ def prepare_video_inputs(
+            height=height,
+            numpify=numpify,
+            torchify=torchify,
+        )
```
I have to move this out of the else block, because `video` is undefined when `equal_resolution` is True.

zucchini-nlp replied:
right, thanks!

jla524 commented:
Hmm... now the modelling tests are failing. I'm not sure if this is the right fix anymore.

jla524 committed:
fix typo at least

ArthurZucker reviewed (approved):
thanks for fixing and making sure we tests a few number of frames!  let's get this merged! 🤗

ArthurZucker commented:
( Checked that `    "tests/models/qwen2_vl/test_image_processing_qwen2_vl.py::Qwen2VLImageProcessingTest::test_video_inputs": {
        "tests_processors": "passed"
    },` passed!
